dandiapi/api/storage.py

```diff
@@ -1,360 +1,20 @@
-        return DandiMinioMultipartManager(self)
-
-    def etag_from_blob_name(self, blob_name) -> str | None:
-        try:
-            response = self.client.stat_object(self.bucket_name, blob_name)
-        except S3Error as e:
-            if e.code == 'NoSuchKey':
-                return None
-            raise
-        else:
-            return response.etag
-
-    def generate_presigned_put_object_url(
-        self, blob_name: str, base64md5: str, tagging: str = ''
-    ) -> str:
-        # Note: minio-py doesn't support using tags or Content-MD5 headers, but we accept these two
-        # parameters to keep parity between the minio and s3 storage functions
-
-        # storage.client will generate URLs like `http://minio:9000/...` when running in
-        # docker. To avoid this, use the secondary base_url_client which is configured to
-        # generate URLs like `http://localhost:9000/...`.
-        return self.base_url_client.presigned_put_object(
-            bucket_name=self.bucket_name,
-            object_name=blob_name,
-            expires=PRESIGNED_URL_TIMEOUT,  # TODO: proper expiration
-        )
-
-    def generate_presigned_head_object_url(self, key: str) -> str:
-        return self.base_url_client.get_presigned_url('HEAD', self.bucket_name, key)
-
-    def generate_presigned_download_url(self, key: str, path: str) -> str:
-        return self.base_url_client.presigned_get_object(
-            self.bucket_name,
-            key,
-            response_headers={'response-content-disposition': f'attachment; filename="{path}"'},
-        )
-
-    def generate_presigned_inline_url(self, key: str, path: str, content_type: str) -> str:
-        return self.base_url_client.presigned_get_object(
-            self.bucket_name,
-            key,
-            response_headers={
-                'response-content-disposition': f'inline; filename="{path}"',
-                'response-content-type': content_type,
-            },
-        )
-
-    def sha256_checksum(self, key: str) -> str:
-        calculator = ChecksumCalculatorFile()
-        obj = self.client.get_object(self.bucket_name, key)
-        for chunk in obj.stream(amt=1024 * 1024 * 16):
-            calculator.write(chunk)
-        return calculator.checksum
-
-
-def create_s3_storage(bucket_name: str) -> Storage:
-    """
-    Return a new Storage instance, compatible with the default Storage class.
-
-    This abstracts over differences between S3Storage and MinioStorage,
-    allowing either to be used as an additional non-default Storage.
-    """
-    # For production, calling django.core.files.storage.get_storage_class is fine
-    # to return the storage class of S3Storage.
-    default_storage_class = get_storage_class()
-
-    if issubclass(default_storage_class, S3Storage):
-        storage = VerbatimNameS3Storage(bucket_name=bucket_name)
-        # Required to upload to the sponsored bucket
-        storage.default_acl = 'bucket-owner-full-control'
```

Comment:
All this ACL stuff is no longer necessary, once https://github.com/dandi/dandi-infrastructure/pull/240 is merged.